gcp-observability/src/main/java/io/grpc/gcp/observability/interceptors/LogHelper.java

```diff
@@ -88,7 +89,9 @@ void logClientHeader(
       GrpcLogRecord.EventLogger eventLogger,
       String callId,
       // null on client side
-      @Nullable SocketAddress peerAddress) {
+      @Nullable SocketAddress peerAddress,
+      // null on server side
```

Comment:
Getting server `SpanContext` in `InternalLoggingServerInterceptor` instead of `TraceLoggingHelper`. 
Updated to use `SpanContext.INVALID` as default SpanContext instead of passing null.